TOOLCHAIN_OPTIONS_append in u-boot_%.bbappend breaks building for other platforms

I have an image that I am building for raspberrypi4-64 for production, and for which I also have a qemux86-64 target for development purposes. To that end, along with the meta-updater layer, I include both this layer and meta-updater-raspberrypi in my project.

It seems like the [TOOLCHAIN_OPTIONS_append in u-boot_%.bbappend](https://github.com/advancedtelematic/meta-updater-qemux86-64/blob/master/recipes-bsp/u-boot/u-boot_%25.bbappend#L11) prevents u-boot from building for raspberrypi4-64 and raspberrypi3-64, though it builds fine for raspberrypi3, raspberrypi4, and qemux86-64.

With this line enabled, building u-boot for raspberrypi4-64 fails because it can't find libgcc:

```
|   aarch64-bantha-linux-ld.bfd    -Ttext 0xc100000 -g -o examples/standalone/hello_world -e hello_world examples/standalone/hello_world.o examples/standalone/libstubs.o -L . -lgcc
| aarch64-bantha-linux-ld.bfd: cannot find -lgcc
```

If I comment the line out, or BBMASK the file out of my build, then u-boot builds successfully.

See https://github.com/agherzan/meta-raspberrypi/issues/766 for more details

--- a/recipes-bsp/u-boot/u-boot_%.bbappend
+++ b/recipes-bsp/u-boot/u-boot_%.bbappend
@@ -8,7 +8,7 @@ SRC_URI +=" \
           "
 
 # fix after default security flags in poky
-TOOLCHAIN_OPTIONS_append = "${SECURITY_NOPIE_CFLAGS}"
+TOOLCHAIN_OPTIONS_append_qemux86-64 = "${SECURITY_NOPIE_CFLAGS}"
 
 do_compile_prepend_qemux86-64 () {
   export BUILD_ROM=y

--- a/recipes-bsp/u-boot/u-boot_%.bbappend
+++ b/recipes-bsp/u-boot/u-boot_%.bbappend
@@ -8,7 +8,7 @@ SRC_URI +=" \
           "
 
 # fix after default security flags in poky
-TOOLCHAIN_OPTIONS_append_qemux86-64 = "${SECURITY_NOPIE_CFLAGS}"
+TOOLCHAIN_OPTIONS_append_qemux86-64 = " ${SECURITY_NOPIE_CFLAGS}"
 
 do_compile_prepend_qemux86-64 () {
   export BUILD_ROM=y

